db/migrate/20201006091457_add_financial_quarter_and_year_to_planned_disbursements.rb

```diff
@@ -0,0 +1,21 @@
+class AddFinancialQuarterAndYearToPlannedDisbursements < ActiveRecord::Migration[6.0]
+  def change
+    add_column :planned_disbursements, :financial_quarter, :integer, index: true
+    add_column :planned_disbursements, :financial_year, :integer, index: true
+
```

Comment:
Do we want to allow `NULL` in those columns?

Reply:
I think I was worried about the data already in production, but looking at it now the migration will populate these values so I can set them to not null - thanks!

Reply:
Yeah, okay - I tried to set this to not allow null, but the columns are added and then then the database complains that we have planned disbursements with null values and it all fails - I guess we would have to ensure the values are set first and then add the constraint after?

How do you think the database constriant differs from the Rails validations - I have these two attributes as required there always?

Reply:
That's interesting. I had a very similar problem with a couple of columns I created on `activities`. They were integers as well and at first I didn't specify the not allow NULL. When I went to change it I couldn't either for the same reason as you, I think.

As you said, we are going to populate the data with a migration so it should be alright. 

About the difference in the constraints from the DB to the validations on the model, I wasn't sure but I've found this thread on StackOverflow that explains the advantages of having both: https://stackoverflow.com/questions/2367281/ruby-on-rails-is-it-better-to-validate-in-the-model-or-the-database 

Having said that, I think we tend to use more the model validation in this project. I haven't seen many `null: false` on our schema. Maybe it's something worth talking about with the wider team?

Reply:
Interesting link! Thank you! I strikes me if this was a completely new table we could do both constraints, but with a change it can be hard as we have to ensure the data is in a fit state before we make the change - that wouldn't be easy here (but not impossible) another one we have to strike the right balance on?

Thanks for calling it out and the discussion.